server/odc-service/src/main/java/com/oceanbase/odc/service/permission/table/TablePermissionService.java

```diff
@@ -0,0 +1,604 @@
+
+    private static final Set<String> ORACLE_DATA_DICTIONARY = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+    private static final Set<String> MYSQL_DATA_DICTIONARY = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+    @Transactional(rollbackFor = Exception.class)
+    @SkipAuthorize("permission check inside")
+    public Page<UserTablePermission> list(@NotNull Long projectId, @NotNull QueryTablePermissionParams params,
+            Pageable pageable) {
+        if (params.getUserId() == null || params.getUserId() != authenticationFacade.currentUserId()) {
+            projectPermissionValidator.checkProjectRole(projectId,
+                    Arrays.asList(ResourceRoleName.OWNER, ResourceRoleName.DBA));
+        } else {
+            projectPermissionValidator.checkProjectRole(projectId, ResourceRoleName.all());
+        }
+        Date expiredTime = new Date(System.currentTimeMillis() - expiredRetentionTimeSeconds * 1000);
+        permissionService.deleteExpiredPermission(expiredTime);
+        Date expireTimeThreshold = TimeUtils.getStartOfDay(new Date());
+        Specification<UserTablePermissionEntity> spec = Specification
+                .where(UserTablePermissionSpec.projectIdEqual(projectId))
+                .and(UserTablePermissionSpec.organizationIdEqual(authenticationFacade.currentOrganizationId()))
+                .and(UserTablePermissionSpec.userIdEqual(params.getUserId()))
+                .and(UserTablePermissionSpec.ticketIdEqual(params.getTicketId()))
+                .and(UserTablePermissionSpec.databaseNameLike(params.getFuzzyDatabaseName()))
+                .and(UserTablePermissionSpec.dataSourceNameLike(params.getFuzzyDataSourceName()))
+                .and(UserTablePermissionSpec.typeIn(params.getTypes()))
+                .and(UserTablePermissionSpec.authorizationTypeEqual(params.getAuthorizationType()))
+                .and(UserTablePermissionSpec.filterByExpirationStatus(params.getStatuses(), expireTimeThreshold));
+        return userTablePermissionRepository.findAll(spec, pageable).map(
+                e -> {
+                    UserTablePermission permission = mapper.entityToModel(e);
+                    Date expireTime = permission.getExpireTime();
+                    if (expireTime.before(expireTimeThreshold)) {
+                        permission.setStatus(ExpirationStatusFilter.EXPIRED);
+                    } else if (expireTime.before(DateUtils.addDays(expireTimeThreshold, EXPIRING_DAYS))) {
+                        permission.setStatus(ExpirationStatusFilter.EXPIRING);
+                    } else {
+                        permission.setStatus(ExpirationStatusFilter.NOT_EXPIRED);
+                    }
+                    return permission;
+                });
+    }
+
+    @Transactional(rollbackFor = Exception.class)
+    @SkipAuthorize("permission check inside")
+    public List<UserTablePermission> listWithoutPage(@NotNull Long projectId,
```

Comment:
This method could be deleted. If you want to query table permission and the result without page, you can invoke `list(projectId, params, Pageable.unpaged()).getContent();`

Reply:
done